### PR TITLE
Tech debt/remove custom abstract main

### PR DESCRIPTION
**Pull request by SchoolGuy**
This is a successor of #227 and part of the split of #210 

We have an abstract custom main class which provides us with nothing but confusion. The static part in the code needs to be cut out elsewhere but for now this removes a lot of unnecessary code.

### CODE REVIEW

**Comment by SchoolGuy** [comment]
Will resolve the conflicts in the evening.

File: src/main/java/menion/android/whereyougo/MainApplication.java [hunk 1]
```diff
@@ -52,7 +52,6 @@
     private static final String TAG = "MainApplication";
 
     private static Timer mTimer;
-    private static OnAppVisibilityChange onAppVisibilityChange;
```
Comment (bekuno) [inline src/main/java/menion/android/whereyougo/MainApplication.java:55]: The changes in this file are independent from the other files.
Please extract this file and create an owh PR.
(e.g. "remove unimplemented code")

File: src/main/java/menion/android/whereyougo/preferences/PreferenceValues.java [hunk 2]
```diff
@@ -193,8 +193,6 @@ public static Activity getCurrentActivity() {
     }
 
     public static void setCurrentActivity(Activity activity) {
-        if (PreferenceValues.currentActivity == null && activity != null)
-            MainApplication.appRestored();
```
Comment (bekuno) [inline src/main/java/menion/android/whereyougo/preferences/PreferenceValues.java:197]: This is result of the change in MainApplication, therefore it should included in their PR.

File: src/main/java/menion/android/whereyougo/gui/activity/CartridgeDetailsActivity.java [hunk 3]
```diff
@@ -86,25 +87,20 @@ public void onCreate(Bundle savedInstanceState) {
...
-                Location loc = new Location(TAG);
-                loc.setLatitude(MainActivity.cartridgeFile.latitude);
-                loc.setLongitude(MainActivity.cartridgeFile.longitude);
-                Guide guide = new Guide(MainActivity.cartridgeFile.name, loc);
-                A.getGuidingContent().guideStart(guide);
-                MainActivity.callGudingScreen(CartridgeDetailsActivity.this);
-                CartridgeDetailsActivity.this.finish();
-                return true;
-            }
+        CustomDialog.setBottom(this, getString(R.string.start), (dialog, v, btn) -> {
```
Comment (bekuno) [inline src/main/java/menion/android/whereyougo/gui/activity/CartridgeDetailsActivity.java:90]: The lambda migration should had to go to #227.
Please do it in a PR with the MainApplication cleanup.

**Comment by SchoolGuy** [comment]
@bekuno Removed commit which contained the additional changes. Is this now ready to be merged? Or do you have more suggestions?

**Comment by bekuno** [comment]
The last commit should go to #237, isn't it?

**Comment by SchoolGuy** [comment]
New PR which will be merged first: #238

**Comment by SchoolGuy** [comment]
@bekuno I rebased this onto master. I will resolve all your comments and would love if you could do another review! :)

File: src/main/java/menion/android/whereyougo/gui/activity/MainActivity.java [hunk 4]
```diff
@@ -399,6 +691,43 @@ public boolean onOptionsItemSelected(MenuItem item) {
         }
     }
 
+    @Override
+    public void onDestroy() {
+        if (finish) {
+            // stop debug if any forgotten
```
Comment (bekuno) [inline src/main/java/menion/android/whereyougo/gui/activity/MainActivity.java:697]: The call sequence is opposite - first the part in (old) MainActivity.onDestroy, there is the call to super.onDestroy()=CustomMainActivity.onDestroy().

Comment (SchoolGuy) [reply src/main/java/menion/android/whereyougo/gui/activity/MainActivity.java]: Addressed it in the force push.

**Comment by SchoolGuy** [comment]
@bekuno I know added a second commit which contains changes which I think would be useful to do as Android Studio is telling me this code is never actually run. If you say so, I will drop it or open a seperate PR,

File: src/main/java/menion/android/whereyougo/gui/activity/MainActivity.java (hunk 4 above)
Comment (bekuno) [reply src/main/java/menion/android/whereyougo/gui/activity/MainActivity.java]: Please check the original code, the "then" part is now necessary.

**Comment by bekuno** [comment]
Please do it in an separate PR.
So it is easier to test.

**Comment by SchoolGuy** [comment]
@bekuno Ready to be merged then I guess? Or any complaints?

File: src/main/java/menion/android/whereyougo/gui/activity/MainActivity.java (hunk 4 above)
Comment (bekuno) [reply src/main/java/menion/android/whereyougo/gui/activity/MainActivity.java]: You need the part
~~~
        } else {
            super.onDestroy();
~~~
see the original implementation.

Comment (SchoolGuy) [reply src/main/java/menion/android/whereyougo/gui/activity/MainActivity.java]: Added

**Comment by bekuno** [comment]
@SchoolGuy  Thanks for your work.

**Pull request merged**